js/demo-apps/packages/react-front-end/src/common/components/SideMenu.tsx

```diff
@@ -122,20 +126,49 @@ export const SideMenu: React.FunctionComponent<{ mode: Mode }> = (props) => {
     exact: true,
   });
 
-  return (
-    <Drawer
-      className={classes.drawer}
-      variant="permanent"
-      classes={{
-        paper: classes.drawerPaper,
-      }}
-    >
+  const theme = useTheme();
+  const onDesktop = useMediaQuery(theme.breakpoints.up('md'));
+
+  const renderResponsiveDrawer = (slot: JSX.Element) => {
+    if (!onDesktop) {
+      return (
+        <Drawer
+          className={classes.drawer}
+          variant="temporary"
+          anchor="left"
+          open={sidebarOpen}
+          onClose={() => setSidebarOpen(false)}
+          classes={{
+            paper: classes.drawerPaper,
+          }}
+          onClick={() => setSidebarOpen(false)} // this should be automatic..but it's not
+        >
+          {slot}
+        </Drawer>
+      );
+    } else {
```

Comment:
Nit: Useless else can be removed here.

I should figure out how to add this to the linter. :thinking: